website_sale_contract/controllers/main.py

```diff
@@ -0,0 +1,123 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import http
+from odoo.http import request
+from odoo.addons.website_sale.controllers.main import WebsiteSale
+
+import json
+
+
+class WebsiteSale(WebsiteSale):
+
+    @http.route(
+        ['/shop/payment'],
+        type='http',
+        auth='public',
+        website=True,
+    )
+    def payment(self, **post):
+
+        Website = request.website.with_context(request.env.context)
+        order = Website.sale_get_order()
+
+        redirection = self.checkout_redirection(order)
+        if redirection:
+            return redirection
+
+        sale_lines = order.sale_line_ids.filtered(
+            lambda s: s.product_id.is_contract
+        )
+
+        sale_lines = sorted(lambda s: s.id, sale_lines)
+
+        if not sale_lines:
+            return super(WebsiteSale, self).payment(**post)
+
+        ContractCheckout = self.env['contract.checkout']
+
+        contract_wizard = ContractCheckout.search(['order_id', '=', order.id])
```

Comment:
You should instead give your wizard model a method that takes an order ID and returns existing or creates new